traffic_ops/testing/api/v3/types_test.go

```diff
@@ -28,224 +34,255 @@ import (
 
 func TestTypes(t *testing.T) {
 	WithObjs(t, []TCObj{Parameters, Types}, func() {
-		GetTestTypesIMS(t)
-		currentTime := time.Now().UTC().Add(-5 * time.Second)
-		time := currentTime.Format(time.RFC1123)
-		var header http.Header
-		header = make(map[string][]string)
-		header.Set(rfc.IfModifiedSince, time)
-		SortTestTypes(t)
-		UpdateTestTypes(t)
-		GetTestTypes(t)
-		GetTestTypesIMSAfterChange(t, header)
+
+		currentTime := time.Now().UTC().Add(-15 * time.Second)
+		currentTimeRFC := currentTime.Format(time.RFC1123)
+		tomorrow := currentTime.AddDate(0, 0, 1).Format(time.RFC1123)
+
+		methodTests := utils.V3TestCase{
+			"GET": {
+				"NOT MODIFIED when NO CHANGES made": {
+					ClientSession:  TOSession,
+					RequestHeaders: http.Header{rfc.IfModifiedSince: {tomorrow}},
+					Expectations:   utils.CkRequest(utils.NoError(), utils.HasStatus(http.StatusNotModified)),
+				},
+				"OK when VALID request": {
+					ClientSession: TOSession,
+					Expectations: utils.CkRequest(utils.NoError(), utils.HasStatus(http.StatusOK), utils.ResponseLengthGreaterOrEqual(1),
+						validateTypeSort()),
+				},
+				"OK when VALID NAME parameter": {
+					ClientSession: TOSession,
+					RequestParams: url.Values{"name": {"ORG"}},
+					Expectations: utils.CkRequest(utils.NoError(), utils.HasStatus(http.StatusOK), utils.ResponseHasLength(1),
+						validateTypeFields(map[string]interface{}{"Name": "ORG"})),
+				},
+			},
+			"POST": {
+				"BAD REQUEST when NOT in server table": {
+					ClientSession: TOSession,
+					RequestBody: map[string]interface{}{
+						"description": "Host header regular expression-Test",
+						"name":        "TEST_1",
+						"useInTable":  "regex",
+					},
+					Expectations: utils.CkRequest(utils.HasError(), utils.HasStatus(http.StatusBadRequest)),
+				},
+				"OK when VALID request in SERVER table": {
+					ClientSession: TOSession,
+					RequestBody: map[string]interface{}{
+						"description": "Host header regular expression-Test",
+						"lastUpdated": "2022-06-17T19:13:46.802044+00:00",
+						"name":        "TEST_4",
+						"useInTable":  "server",
+					},
+					Expectations: utils.CkRequest(utils.NoError(), utils.HasStatus(http.StatusOK),
+						validateTypeUpdateCreateFields("TEST_4", map[string]interface{}{"Name": "TEST_4"})),
+				},
+			},
+			"PUT": {
+				"BAD REQUEST when NOT in server table": {
```

Comment:
nit: Can we change this to: "BAD REQUEST when USEINTABLE NOT server". Same for POST and v4